scipy/stats/_distn_infrastructure.py

```diff
@@ -2230,6 +2279,33 @@ def _nnlf_and_penalty(self, x, args):
             return -np.sum(logpdf[finite_logpdf], axis=0) + penalty
         return -np.sum(logpdf, axis=0)
 
+    def _censored_nnlf_and_penalty(self, x, args):
+        # Modified version of _nnlf_and_penalty.
+        cond0 = ~self._support_mask(x, *args)
+        n_bad = np.count_nonzero(cond0)
+        if n_bad > 0:
+            x = CensoredData(x._lower[~cond0], x._upper[~cond0])
+
+        # logpdf of the noncensored data.
+        logpdf_nc = self._logpdf(x._lower[x._not_censored], *args)
```

Comment:
`_nnlf_and_penalty` takes care with `finite_logpdf = np.isfinite(logpdf)`;  n_bad += np.sum(~finite_logpdf, axis=0). Why don't we have to do that here?

Reply:
This has been changed in the update.